tidy3d/components/simulation.py

```diff
@@ -2579,7 +2595,10 @@ def make_eps_data(coords: Coords):
             return xr.DataArray(eps_array, coords=coords, dims=("x", "y", "z"))
 
         # combine all data into dictionary
-        coords = grid[coord_key]
+        if coord_key[0] != "E":
```

Comment:
maybe a bit simpler to do
```py
if coord_key[0] == "E":
    coord_key = coord_key[0:2]
coords = grid[coord_key]
return make_eps_data(coords)
```
although I',m not exactly sure what this bit is doing exactly.

Reply:
I changed this part a little bit to try to make it more clear.
Here I'm trying to generalize `epsilon_on_grid(..., coord_key: str, ...)` to be able to return off-diagonal components too. The way I did this is that in addition to `coord_key = "Ex", "Ey", "Ez"` we can also call `coord_key = "Exy", "Exz", "Eyx", ...` which would return a corresponding off-diagonal component but still at an Yee location (e.g., `eps_xy` at `Ex`). So, for those keys I just remove the last letter.

What I'm a bit hesitant about is that: 
- things like `Ezy` are not proper physical concepts
- from efficiency point of view, maybe it would make more sense to add a separate flag `include_off_diag`, and then `coord_key="Ex", include_off_diag=True` would return three arrays `eps_xx, eps_xy, eps_xz`